src/Utils/ProjectsUtils.ts

```diff
@@ -31,7 +31,18 @@ export async function resolveProjectNameByIssueId(teamscaleClient: TeamscaleClie
                 if (testGapSummary && testGapSummary.numberOfChangedMethods > 0) {
                     return projectCandidate;
                 }
-            } else {
+            }
+            else if (badgeType === BadgeType.TestSmell) {
```

Comment:
This method (resolveProjectNameByIssueId) could be refactored to be a bit easier to maintain:
we could do something like
```
    for (const projectCandidate of projectCandidates) {
        let isMatch = false;
        try {
            if (badgeType === BadgeType.TestGap) {
               isMatch = handleTestGapBadge(projectCandidate);
            }
            else if (badgeType === BadgeType.TestSmell) {
               isMatch = handleTestSmellBadge(projectCandidate);
           } else {
              isMatch = handleFindingsBadge(projectCandidate);  
           }
           if (isMatch) { return projectCandidate;}
        }
        ...
        ...
    }
```
then we could have the code in methods to test whether the project is a match.
WDYT?

Reply:
Good suggestion. Since the methods have to be async, they must return a promise. That's why I had to enhance your suggestion and 'await' each of these methods and then check the variable :)